### DISCUSSION TITLE
Restart container services from Machinaris console

### DISCUSSION THREAD

**Discussion by LeroyINC** [guydavis/machinaris#728] [General]
Guy,

Congrats on your recent releases you have made some very nice improvements and I thank you for that.

Question.. is there a way to restart the container services from the web console? sometime chia or whatever fork gets gummed up and then I have to connect to whatever device its running on and either reboot it or stop the container and start it again.

would be nice if there could be a 1 click button in the workers page or, where ever you think is best, to restart the Chia services.

another example of where this would be handy.. is when modifying the config.yaml file from inside the Machinaris console.... 

Thanks.. and keep up the good work

**Comment by guydavis** [comment]
Hi, yes I may need to enhance this in the WebUI.  Currently the following is already happening behind the scenes so much of your request is in place today.

1. When you visit Settings | Farming page, edit the config file contents, and hit Save button.  -> The blockchain farming is [automatically restarted](https://github.com/guydavis/machinaris/blob/main/api/commands/chia_cli.py#L71) as per message shown.
2. Machinaris is continually watching the blockchain sync height to ensure it keeps increasing.  If it sees that the peak height gets stuck for 15 minutes, then it will [execute a farming restart](https://github.com/guydavis/machinaris/blob/main/api/schedules/restart_stuck_farmer.py#L50) as shown in the Server Log.

Both features have been in place for many months and have helped to keep my blockchains synced in my experience on my Unraid server.  That said, these background behaviors are not particularly visible to the user, so they many not know this is happening.  

I'll give this request to place restart functionality into the WebUI directly a look when I get a chance.  Thanks for the suggestion.

**Comment by LeroyINC** [comment]
Guy,

I don’t recall it auto restarting when the config file is changed.
I have seen this behavior when the fork tools settings are changed.

Does Machinaris watch for wallet height to keep increasing and restart?
I have seen an issue where the wallet got stuck and I had to restart the docker container


Thank You,
Leroy Ladyzhensky

440-937-7680
